### PR TITLE
Networking

### PR DESCRIPTION
**Pull request by yopyop1007**
Merging in the networking gRPC stuff we've been working on. This includes the common proto file that every subgroup will add their RPCs/messages to - this protofile is called rbox.proto and is located in src/main/proto. There are also a simple example included that defines the connectTo RPC. The photo file for this example is called network.proto and is located in the same place as rbox.proto. The gRPC-generated Java class files for network.proto are named ConnectionServiceGrpc and Network, and are located in "target/generated-sources/protobuf/grpc-java/edu/rice/comp413fall2020grey/Networking" and "target/generated-sources/protobuf/java/edu/rice/comp413fall2020grey/Networking" respectively. The implementations of the example (a gRPC service implementation, server implementation, and client implementation) are in the package Networking, which is in "/Users/vikramnayar/IdeaProjects/RBox/core/src/edu/rice/comp413fall2020grey/Networking".

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,28 +25,62 @@
             <version>2.21</version>
```
Comment (popatr15) [inline pom.xml:37]: Can you fix the conflict in this file

**Review by popatr15 — CHANGES_REQUESTED** [review]
Please fix the Merge conflict in the pom.xml. Also pls pull the recent changes from the git ignore so that we do not include binaries in the repository.

File: src/main/proto/network.proto [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+message ConnectionConfirmation {
+  string confirmation = 1;
+}
+
+// Defining a Service, a Service can have multiple RPC operations
+service ConnectionService {
+  // Define a RPC operation
+  rpc connectTo(ServerInformation) returns (ConnectionConfirmation);
```
Comment (isabella-h) [inline src/main/proto/network.proto:28]: Shall we merge the messages and service defined in `network.proto` into `rbox.proto`?

Comment (yopyop1007) [reply src/main/proto/network.proto:28]: No, I think we can remake those anyways. the network.proto stuff is just there to serve as an example

Comment (isabella-h) [reply src/main/proto/network.proto:28]: Then may I remove `rpc connectTo(ServerInformation) returns (ConnectionConfirmation);` in `rbox.proto`? It's giving me error `"ServerInformation" is not defined. "ConnectionConfirmation" is not defined.` when I tried to compile.

Comment (yopyop1007) [reply src/main/proto/network.proto:28]: Yeah, you can delete all the network.proto stuff from rbox.proto

**Review by popatr15 — APPROVED** [review]

**Review by mja9 — APPROVED** [review]

**Pull request merged**